build.yaml

```diff
@@ -1441,6 +1441,35 @@ steps:
       - make_docs
     clouds:
       - gcp
+  - kind: runImage
+    name: build_native_query
+    image:
+      valueFrom: hail_build_image.image
+    script: |
+      set -ex
+      export LLVM_DIR=/usr/lib/llvm-15
+      cd /io/repo/query
+      mkdir build
+      cd build
+      cmake .. \
+        -G Ninja \
+        -DCMAKE_C_COMPILER=$LLVM_DIR/bin/clang \
+        -DCMAKE_CXX_COMPILER=$LLVM_DIR/bin/clang++ \
+        -DCMAKE_C_FLAGS=-Werror \
+        -DCMAKE_CXX_FLAGS=-Werror \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+        -DMLIR_DIR=$LLVM_DIR/lib/cmake/mlir
+        # -DLLVM_BUILD_BINARY_DIR=$LLVM_SRC_DIR/build/bin # FIXME need llvm-lit for tests
```

Comment:
See first comment. This would ultimately be necessary as a layer in a build docker image, right?

Reply:
I don't think so. Can you clarify? This is just to build the native code in CI to make sure no one breaks the build as we develop. I expect it to be expanded upon soon as we get the test suite working and built up.

Reply:
Nvm as I'm re-reading it this makes sense.